Activities/Cryptography/UiPath.Cryptography/CryptographyHelper.cs

```diff
@@ -260,5 +315,145 @@ private static int[] GetLegalKeySizes(SymmetricAlgorithm algorithm)
+            Buffer.BlockCopy(inputBytes, salt.Length + iv.Length, encryptedData, 0, encryptedData.Length);
+            Buffer.BlockCopy(inputBytes, salt.Length + iv.Length + encryptedData.Length, tag, 0, tag.Length);
+
+            using (Rfc2898DeriveBytes pbkdf2 = new Rfc2898DeriveBytes(key, salt, PBKDF2_Iterations))
+            {
+                var Key = pbkdf2.GetBytes(32); //256 bit key
+
+                var aes = new AesGcm(Key);
+                decrypted = new byte[encryptedData.Length];
+                aes.Decrypt(iv, encryptedData, tag, decrypted);
+            }
+
+            return decrypted;
+        }
+#endif
+
+#if NET461
+
+        private static byte[] EncryptAesGcmNet461(byte[] inputBytes, byte[] key, out byte[] result)
+        {
+            byte[] encrypted;
+            byte[] salt = new byte[PBKDF2_SaltSizeBytes];
+            var tag = new byte[16];
+            var algorithmIV = new byte[12];
+
+            _rng.GetBytes(salt);
+            _rng.GetBytes(algorithmIV);
+
+            using (Rfc2898DeriveBytes pbkdf2 = new Rfc2898DeriveBytes(key, salt, PBKDF2_Iterations))
+            {
+                var Key = pbkdf2.GetBytes(32); //256 bit key
+                encrypted = EncryptWithBouncyCastle(inputBytes, Key, algorithmIV, out tag);
+            }
+
+            result = new byte[salt.Length + algorithmIV.Length + encrypted.Length + tag.Length];
+            Buffer.BlockCopy(salt, 0, result, 0, salt.Length);
+            Buffer.BlockCopy(algorithmIV, 0, result, salt.Length, algorithmIV.Length);
+            Buffer.BlockCopy(encrypted, 0, result, salt.Length + algorithmIV.Length, encrypted.Length);
+            Buffer.BlockCopy(tag, 0, result, salt.Length + algorithmIV.Length + encrypted.Length, tag.Length);
+
+            return result;
+        }
+
+        private static byte[] DecryptAesGcmNet461(byte[] inputBytes, byte[] key, out byte[] decrypted)
+        {
+            byte[] salt = new byte[PBKDF2_SaltSizeBytes];
+            byte[] iv = new byte[12];
+            byte[] tag = new byte[16];
+            byte[] encryptedData = new byte[inputBytes.Length - salt.Length - iv.Length - tag.Length];
+
+            Buffer.BlockCopy(inputBytes, 0, salt, 0, salt.Length);
+            Buffer.BlockCopy(inputBytes, salt.Length, iv, 0, iv.Length);
+            Buffer.BlockCopy(inputBytes, salt.Length + iv.Length, encryptedData, 0, encryptedData.Length);
+            Buffer.BlockCopy(inputBytes, salt.Length + iv.Length + encryptedData.Length, tag, 0, tag.Length);
+
+            using (Rfc2898DeriveBytes pbkdf2 = new Rfc2898DeriveBytes(key, salt, PBKDF2_Iterations))
+            {
+                var Key = pbkdf2.GetBytes(32); //256 bit key
+                decrypted = DecryptWithBouncyCastle(encryptedData, iv, tag, Key);
+            }
+
+            return decrypted;
+        }
+
+        private static byte[] EncryptWithBouncyCastle(byte[] plaintext, byte[] key, byte[] nonce, out byte[] tag)
```

Comment:
Let's rename this to `EncryptAesGsmWithBouncyCastle` or similar, so that future us will know what algorithm the code uses.

Same for decrypt.

Reply:
renamed